kotlinx-coroutines-test/common/src/TestScope.kt

```diff
@@ -84,6 +85,13 @@ public fun TestScope.runCurrent(): Unit = testScheduler.runCurrent()
 @ExperimentalCoroutinesApi
 public fun TestScope.advanceTimeBy(delayTimeMillis: Long): Unit = testScheduler.advanceTimeBy(delayTimeMillis)
 
+/**
+ * The [test scheduler][TestScope.testScheduler] as a [TimeSource].
+ * @see TestCoroutineScheduler.timeSource
+ */
+@ExperimentalTime
+public val TestScope.testTimeSource: TimeSource get() = testScheduler.timeSource
```

Comment:
Shouldn't it be `timeSource` or `schedulerTimeSource`?
Also lacks `ExperimentalCoroutinesApi`

Reply:
It was initially `timeSource`, but I advised against it because seeing just a bare `timeSource` in the middle of a test will be disheartening, I assume. `schedulerTimeSource` is not much better than `testScheduler.timeSource`, so I think it's either `testTimeSource` or no extension on `TestScope` at all.

Reply:
Added annotation as well